### ISSUE TITLE
Fresh Python 3.12 Install Not Allowed

### ISSUE THREAD

**Issue by ba05** [spyder-ide/spyder#21519]
I created a fresh Python 3.12 environment with miniforge:
`mamba create -n env312 python=3.12`
 Then I ran:
 `mamba install spyder=5.5.0`

And got:
```
Could not solve for environment specs
The following packages are incompatible
└─ spyder 5.5.0**  is installable with the potential options
   ├─ spyder 5.5.0 would require
   │  └─ python >=3.10,<3.11.0a0 , which can be installed;
   ├─ spyder 5.5.0 would require
   │  └─ python >=3.11,<3.12.0a0 , which can be installed;
   ├─ spyder 5.5.0 would require
   │  └─ python >=3.8,<3.9.0a0 , which can be installed;
   └─ spyder 5.5.0 would require
      └─ python >=3.9,<3.10.0a0 , which can be installed.
```
It appears spyder is not allowing Python 3.12?

Package List:
```
(env312) C:\Users\userX>conda list
# packages in environment at C:\Users\userX\AppData\Local\miniforge3\envs\env312:
#
# Name                    Version                   Build  Channel
bzip2                     1.0.8                hcfcfb64_5    conda-forge
ca-certificates           2023.7.22            h56e8100_0    conda-forge
libexpat                  2.5.0                h63175ca_1    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libsqlite                 3.44.0               hcfcfb64_0    conda-forge
libzlib                   1.2.13               hcfcfb64_5    conda-forge
openssl                   3.1.4                hcfcfb64_0    conda-forge
pip                       23.3.1             pyhd8ed1ab_0    conda-forge
python                    3.12.0          h2628c8c_0_cpython    conda-forge
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5226925_1    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
vc                        14.3                h64f974e_17    conda-forge
vc14_runtime              14.36.32532         hdcecf7f_17    conda-forge
vs2015_runtime            14.36.32532         h05e6639_17    conda-forge
wheel                     0.41.3             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge

```

**Comment by dalthviz** [comment]
Hi @ba05 thank you for the feedback! I think the problem is that the conda packages for Spyder 5.5.0 with Pyhton 3.12 doesn't exist yet. I think at some point conda-forge will start building packages for Python 3.12 and stop building for Python 3.8 but for the moment Spyder is only available for Python [3.8, 3.9, 3.10 and 3.11 through conda-forge](https://anaconda.org/conda-forge/spyder/files?version=5.5.0). I think you can check the conda-forge migration status for Python 3.12 packages at https://conda-forge.org/status/#python312

If you want to check Spyder with Python 3.12 I think you can, as you did, create an env with Python 3.12 and try to install Spyder (and any other packages you need to use) with pip. Although this is not the recommended way to install Spyder, and mixing pip with conda packages can cause troubles, as long as you use pip to install anything in the env that should decrease chances for it to break. 

Besides that, another way to use Python 3.12 with Spyder will be to [use the custom interpreter option](https://docs.spyder-ide.org/5/faq.html#using-existing-environment) (since `spyder-kernels` is kind of a `noarch` package I think you should be able to install it on an env with Python 3.12 🤔)

Let us know if the info above helps!

**Comment by ba05** [comment]
Thanks I think the pip method should work in the interim. I was not able to determine the status or where the hold up was for the migration to 3.12. Not sure if there is a web link that shows the package status and where it is in queue? Is it possible it is a spyder dependency that is the hold up?

**Comment by dalthviz** [comment]
> Not sure if there is a web link that shows the package status and where it is in queue? Is it possible it is a spyder dependency that is the hold up?

Not totally sure if there is a more detailed page than the conda-forge status one linked above, but from that link, clicking the `NOT-SOLVABLE` button and then searching with my browser for `spyder` I see the following:

![image](https://github.com/spyder-ide/spyder/assets/16781833/347b5d14-b8bb-40c7-9f08-a52da1e01f7a)
 
So seems like the migration is hold up due to an error with `pyobjc-framework-coreservices` 🤔

**Comment by ccordoba12** [comment]
Thanks for checking @dalthviz! I opened PR https://github.com/conda-forge/pyobjc-framework-coreservices-feedstock/pull/13 that should unblock the migration to Python 3.12

**Comment by ccordoba12** [comment]
This is fixed now. See https://github.com/conda-forge/spyder-feedstock/pull/152

So, Spyder should be available for Python 3.12 in Conda-forge in a couple of hours.

**Comment by w-mwita** [comment]
So meaning if python 3.12 is already installed it would be so difficult to install spyder from the command line?
